From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct hw_perf_event
Patch-mainline: never, kabi

In 4.4.21, commit 0b152db0426acee35ae4798f682ce512fc7f6a35
(perf/x86/cqm: Fix CQM handling of grouping events into a
cache_group), upstream commit a223c1c7ab4cc64537dc4b911f760d851683768a
added one member to struct hw_perf_event. It indeed changed the layout
of that structure.

But the size of struct hw_perf_event did not change as the member
was added to a union and other members of the union are larger (by
at least 20 bytes). And since struct hw_perf_event is allocated as a
part of struct perf_event in perf_event_alloc, we can simply ignore
the change, as nothing actually changed for others. So protect the
change by __GENKSYMS__ and be done with it.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/linux/perf_event.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/include/linux/perf_event.h
+++ b/include/linux/perf_event.h
@@ -121,7 +121,9 @@ struct hw_perf_event {
 		struct { /* intel_cqm */
 			int			cqm_state;
 			u32			cqm_rmid;
+#ifndef __GENKSYMS__
 			int			is_group_event;
+#endif
 			struct list_head	cqm_events_entry;
 			struct list_head	cqm_groups_entry;
 			struct list_head	cqm_group_entry;
